pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -708,6 +741,123 @@ func initInfraMachineTemplateAndInfraClusterFromProvider(platform configv1.Platf
 	}
 }
 
+func (r *MachineSetSyncReconciler) reconcileMAPItoCAPIMachineSetDeletion(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, capiMachineSet *capiv1beta1.MachineSet) (bool, error) {
+	logger := log.FromContext(ctx)
+
+	if mapiMachineSet.DeletionTimestamp.IsZero() {
+		if capiMachineSet == nil || capiMachineSet.DeletionTimestamp.IsZero() {
+			// Neither MAPI authoritative machine set nor its CAPI non-authoritative mirror
+			// are being deleted, nothing to reconcile for deletion.
+			return false, nil
+		}
+
+		// The MAPI authoritative machine set is not being deleted, but the CAPI non-authoritative one is.
+		// Issue a deletion also to the MAPI authoritative machine set.
+		logger.Info("The non-authoritative Cluster API machine set is being deleted, issuing deletion to the corresponding Machine API machine set")
```

Comment:
Does the logger we're making here have the machine set's name in a structured field?